src/main/cpp/states/intake/IntakeStateMgr.cpp

```diff
@@ -113,23 +113,18 @@ IntakeStateMgr::IntakeStateMgr() : m_stateVector(),
 
 /// @brief  run the current state
 /// @return void
-void IntakeStateMgr::RunCurrentState()
+void IntakeStateMgr::RunCurrentState( INTAKE_STATE intake_state)
```

Comment:
RunCurrentState doesn't need to take in a  state.  That is what SetCurrentState is for.  In SetCurrentState, it should save the state in m_currentState, so now you run this state.